security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/action/GeneratorActionDeserializer.java

```diff
@@ -0,0 +1,99 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.json.action;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.security.action.GeneratorAction;
+import com.powsybl.security.action.GeneratorActionBuilder;
+
+import java.io.IOException;
+
+/**
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class GeneratorActionDeserializer extends StdDeserializer<GeneratorAction> {
+
+    public GeneratorActionDeserializer() {
+        super(GeneratorAction.class);
+    }
+
+    private static class ParsingContext {
+        String id;
+        String generatorId;
+        Boolean activePowerRelativeValue;
+        Double activePowerValue;
+        Boolean voltageRegulatorOn;
+        Double targetV;
+        Double targetQ;
+    }
+
+    @Override
+    public GeneratorAction deserialize(JsonParser parser, DeserializationContext deserializationContext) throws IOException {
+        ParsingContext context = new ParsingContext();
+        JsonUtil.parsePolymorphicObject(parser, name -> {
+            switch (name) {
+                case "type":
+                    if (!GeneratorAction.NAME.equals(parser.nextTextValue())) {
+                        throw JsonMappingException.from(parser, "Expected type " + GeneratorAction.NAME);
+                    }
+                    return true;
+                case "id":
+                    context.id = parser.nextTextValue();
+                    return true;
+                case "generatorId":
+                    context.generatorId = parser.nextTextValue();
+                    return true;
+                case "activePowerRelativeValue":
+                    parser.nextToken();
+                    context.activePowerRelativeValue = parser.getValueAsBoolean();
+                    return true;
+                case "activePowerValue":
+                    parser.nextToken();
+                    context.activePowerValue = parser.getValueAsDouble();
+                    return true;
+                case "voltageRegulatorOn":
+                    parser.nextToken();
+                    context.voltageRegulatorOn = parser.getValueAsBoolean();
+                    return true;
+                case "targetV":
+                    parser.nextToken();
+                    context.targetV = parser.getValueAsDouble();
+                    return true;
+                case "targetQ":
+                    parser.nextToken();
+                    context.targetQ = parser.getValueAsDouble();
+                    return true;
+                default:
+                    return false;
+            }
+        });
+        GeneratorActionBuilder generatorActionBuilder = new GeneratorActionBuilder();
```

Comment:
not important but we could have use the builder straight during the parsing, saving the creation of the parsing context object